### DISCUSSION TITLE
Unable to login with wallet

### DISCUSSION THREAD

**Discussion by PolEpie** [WalletConnect/walletconnect-monorepo#1718] [Developer Support]
hi, i've tried to tweak the dapp exemple to use rainbow or trust as wallet like this
<img width="666" alt="CleanShot 2022-12-05 at 22 11 26@2x" src="https://user-images.githubusercontent.com/36932762/205743571-c8116a16-c388-430e-851b-06459f99f704.png">

but i got error when connecting even if the uri looks good

**Comment by PolEpie** [comment]
And when i put core:// as url it open core but doesn't display nothing

**Comment by reslear** [comment]
same, can help ?

**Comment by chadyj** [comment]
Great if you can add more debug info:


**SDK Version**
 - Client: [e.g.  Swift]
 - Version [e.g. 22]

**Reply by PolEpie** [reply]
Client swift last version

**Reply by chadyj** [reply]
What Swift client? There are multiple.

**Reply by PolEpie** [reply]
v2 https://github.com/WalletConnect/WalletConnectSwiftV2

I've used the exemple dapp

**Reply by reslear** [reply]
@chadyj can help?

**Reply by nomadcalendar** [reply]
also looking for a solution to this - @chadyj would appreciate your help!

**Reply by chadyj** [reply]
Discussing this internally. We think we have a fix.

**Comment by flypaper0** [comment]
@PolEpie thanks for reaching us. 

I made some tests and find the way how to make it work. Need to use rfc3986 to encode query params. 

```swift 
extension CharacterSet {
   static var rfc3986: CharacterSet {
        return .alphanumerics.union(CharacterSet(charactersIn: "-._~"))
    }
}

let encoded = uri.addingPercentEncoding(withAllowedCharacters: .rfc3986)!
let url = URL(string: "\(wallet.universal)/wc?uri=\(encoded)")!
```

**Reply by flypaper0** [reply]
You can use it as temp fix, I think we should include this in our SDK

**Reply by PolEpie** [reply]
Ok thanks ! Will try it

**Reply by PolEpie** [reply]
So i've tried with this code 

```swift
@objc func connectWithExampleWallet() {
        let encoded = uri.absoluteString.addingPercentEncoding(withAllowedCharacters: .rfc3986)!
        print("core://wc?uri=\(encoded)")
        let url = URL(string: "core://wc?uri=\(encoded)")!
        DispatchQueue.main.async {
            UIApplication.shared.open(url, options: [:], completionHandler: nil)
        }
 }
```

So URI used to connect look like that 

`core://wc?uri=wc%3Affea0f2ca6859c5c3cf82d60644fa44e37452dd7100eab8b55731e6a344d0e07%402%3FsymKey%3Ddb411f86f3198dc3e82b1557608890c10e468d257ac0cbdd2818b05fd2848757%26relay-protocol%3Dirn`


But still the same issue nothing going on, just Core that open

**Reply by PolEpie** [reply]
It seem work on BitKeep, weird cause core was marked as V2 Wallet

**Reply by flypaper0** [reply]
does it works with TrustWallet and "trust" schema?

**Reply by PolEpie** [reply]
Yes

**Reply by flypaper0** [reply]
Try to pair https://react-app.walletconnect.com with BitKeep. If it doesn't work, that means what BitKeep has an integration problems

**Reply by PolEpie** [reply]
No i mean bitkeep worked but not core

**Reply by flypaper0** [reply]
Also I suggest to use universal links instead of schema
